modules/MatchProvider.js

```diff
@@ -17,34 +18,35 @@ class MatchProvider extends React.Component {
     // **IMPORTANT** we must mutate matches, never reassign, in order for
     // server rendering to work w/ the two-pass render approach for Miss
     this.matches = []
```

Comment:
It looks like server-rendering does work at the moment, that test is correct. The `matches` in the current version is an array of objects for **matching** `<Match>`'s, whereas you're making it an array of functions, 1 from `<Match>` component?

Did you mean this now breaks with the changes in this PR?